Add routing to app

nadzeyamolchan opened the pull request.

nadzeyamolchan committed:
add theme,
add header component,
add footer component,
fix pagination buttons,
create directory for constants

nadzeyamolchan committed:
fix style,
add homepage,
add page for pokemon table

nadzeyamolchan committed:
add routing

nadzeyamolchan committed:
add pokemon-card page

nadzeyamolchan committed:
fix: add semantic tag
fix: remove setTime out (for checking skeleton)

hannamaskalenka commented on src/PokemonApp.js:
```diff
@@ -1,10 +1,27 @@
+                <Switch>
+                    <Route exact path='/' component={Homepage}/>
+                    <Route exact path='/pokemon' component={PokemonTablePage}/>
+                    <Route path='/pokemon/:name' component={PokemonCardPage}/>
```
Probably it would be better to use _id_ as param

hannamaskalenka commented on src/PokemonApp.js:
```diff
@@ -1,10 +1,27 @@
-import PokemonsTable from "./components/table/PokemonsTable.component";
+
 import {ErrorBoundary} from "./error/ErrorBoundary";
+import {theme} from "./theme";
```
Please make correct order of imports

hannamaskalenka commented on src/components/pokemon-card/pokemon-card.component.jsx:
```diff
@@ -0,0 +1,33 @@
+import './pokemon-card.style.css';
+
+const PokemonCard = (props) => {
+    console.log(props);
```
Please remove console.log

hannamaskalenka commented on src/components/pokemon-card/pokemon-card.component.jsx:
```diff
@@ -0,0 +1,33 @@
+export default PokemonCard;
+
+/*
+<div>
```
Please remove unnecessary comments

nadzeyamolchan committed:
change namings of files,
create styles using material ui instead of pure css,
remove unnecessary 'console.log'

hannamaskalenka commented on src/pages/PokemonCardPage/PokemonCardPage.jsx:
```diff
@@ -0,0 +1,26 @@
+                         id={pokemon.id}
+                         height={pokemon.height}
+                         weight={pokemon.weight}
+                         image={pokemon.sprites.other["official-artwork"].front_default}
```
Probably it would be better to create a constant here for every name of sprites or something like that and move it to constants file

hannamaskalenka commented on src/error/ErrorBoundary.js:
```diff
@@ -21,7 +21,7 @@ export class ErrorBoundary extends Component {
                 <div>
                     <h2>Something went wrong.</h2>
-                    {<details style={{whiteSpace: 'pre-wrap'}}>
+                    {<details>
```
Unnecessary curly brackets

hannamaskalenka commented on src/components/PokemonTable/PokemonTable.component.jsx:
```diff
@@ -0,0 +1,129 @@
+                                        <TableCell className={classes.cell}>
+                                            <Link to={'/pokemon/' + pokemon.name}>
+                                                <figure>
+                                                    <img src={pokemon.sprites.other['official-artwork'].front_default}
```
Same about constant for name of sprites

nadzeyamolchan committed:
change makeStyles function,
change footer,header, and pokemon table styles,
replace container tag with semantic in footer and header

nadzeyamolchan committed:
change path param to 'id' instead 'name'

nadzeyamolchan committed:
change import order

nadzeyamolchan committed:
create constant for name of sprite
